python/cugraph/cugraph/gnn/data_loading/dist_sampler.py

```diff
@@ -0,0 +1,577 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import warnings
+from math import ceil
+
+import pylibcugraph
+import numpy as np
+import cupy
+import cudf
+
+from typing import Union, List, Dict, Tuple
+from cugraph.utilities import import_optional
+from cugraph.gnn.comms import cugraph_comms_get_raft_handle
+
+from cugraph.gnn.data_loading.bulk_sampler_io import create_df_from_disjoint_arrays
+
+# PyTorch is NOT optional but this is required for container builds.
+torch = import_optional("torch")
+
+TensorType = Union[torch.Tensor, cupy.ndarray, cudf.Series]
```

Comment:
Looks like this forces `torch` to be required instead of optional. We're seeing this in CI:
```
/opt/conda/envs/test/lib/python3.9/site-packages/cugraph/gnn/data_loading/dist_sampler.py:32: in <module>
    TensorType = Union[torch.Tensor, cupy.ndarray, cudf.Series]
/opt/conda/envs/test/lib/python3.9/site-packages/cugraph/utilities/utils.py:410: in __getattr__
    raise RuntimeError(f"This feature requires the {self.name} " "package/module")
E   RuntimeError: This feature requires the torch package/module
```
We should either update `import_optional` to allow it to support importing objects from modules, or maybe just treat `torch.Tensor` like a `ForwardRef` type by adding quotes?
```suggestion
TensorType = Union["torch.Tensor", cupy.ndarray, cudf.Series]
```